dynawo/sources/Models/CPP/ModelLoadRestorativeWithLimits/DYNModelLoadRestorativeWithLimits.cpp

```diff
@@ -0,0 +1,579 @@
+    ++localIndex;
+    QYNum_ = localIndex;
+  }
+
+  void
+  ModelLoadRestorativeWithLimits::collectSilentZ(BitMask* silentZTable) {
+    silentZTable[0].setFlags(NotUsedInDiscreteEquations);
+    silentZTable[1].setFlags(NotUsedInDiscreteEquations);
+  }
+
+  void
+  ModelLoadRestorativeWithLimits::getSize() {
+    if (getIsInitProcess()) {
+      sizeF_ = 0;
+      sizeY_ = 0;
+      sizeZ_ = 0;
+      sizeG_ = 0;
+      sizeMode_ = 0;
+    } else {
+      sizeF_ = 6;
+      sizeY_ = 8;
+      sizeZ_ = 2;
+      sizeG_ = 2;
+      sizeMode_ = 1;
+      calculatedVars_.assign(nbCalculatedVariables_, 0);
+    }
+  }
+
+  void
+  ModelLoadRestorativeWithLimits::evalYType() {
+    unsigned int yTypeIndex = 0;
+    yType_[yTypeIndex] = DIFFERENTIAL;  // differential equation on ufRaw
+    ++yTypeIndex;
+    yType_[yTypeIndex] = ALGEBRAIC;  // uf
+    ++yTypeIndex;
+    yType_[yTypeIndex] = EXTERNAL;  // ur
+    ++yTypeIndex;
+    yType_[yTypeIndex] = EXTERNAL;  // ui
+    ++yTypeIndex;
+    yType_[yTypeIndex] = ALGEBRAIC;  // ir
+    ++yTypeIndex;
+    yType_[yTypeIndex] = ALGEBRAIC;  // ii
+    ++yTypeIndex;
+    yType_[yTypeIndex] = ALGEBRAIC;  // p
+    ++yTypeIndex;
+    yType_[yTypeIndex] = ALGEBRAIC;  // q
+  }
+
+  void
+  ModelLoadRestorativeWithLimits::evalFType() {
+    fType_[0] = DIFFERENTIAL_EQ;  // differential equations
+    fType_[1] = ALGEBRAIC_EQ;
+    fType_[2] = ALGEBRAIC_EQ;
+    fType_[3] = ALGEBRAIC_EQ;
+    fType_[4] = ALGEBRAIC_EQ;
+    fType_[5] = ALGEBRAIC_EQ;
+  }
+
+  double
+  ModelLoadRestorativeWithLimits::UfRaw() const {
+    return yLocal_[UfRawYNum_];
+  }
+
+  double
+  ModelLoadRestorativeWithLimits::Uf() const {
+    return yLocal_[UfYNum_];
+  }
+
+  double
+  ModelLoadRestorativeWithLimits::UfRawPrim() const {
+    return ypLocal_[UfRawYNum_];
+  }
+
+  double
+  ModelLoadRestorativeWithLimits::Ur() const {
+    return yLocal_[UrYNum_];
+  }
+
+  double
+  ModelLoadRestorativeWithLimits::Ir() const {
+    return yLocal_[IrYNum_];
+  }
+
+  double
+  ModelLoadRestorativeWithLimits::Ii() const {
+    return yLocal_[IiYNum_];
+  }
+
+  double
+  ModelLoadRestorativeWithLimits::Ui() const {
+    return yLocal_[UiYNum_];
+  }
+
+  double
+  ModelLoadRestorativeWithLimits::U() const {
+    return sqrt(Ur() * Ur() + Ui() * Ui());
+  }
+
+  double
+  ModelLoadRestorativeWithLimits::U2() const {
+    return Ur() * Ur() + Ui() * Ui();
+  }
+
+  double
+  ModelLoadRestorativeWithLimits::P() const {
+    return yLocal_[PYNum_];
+  }
+
+  double
+  ModelLoadRestorativeWithLimits::Q() const {
+    return yLocal_[QYNum_];
+  }
+
+  void
+  ModelLoadRestorativeWithLimits::evalF(double /*t*/, propertyF_t type) {
+    if (getIsInitProcess())
+      return;
+    if (isConnected()) {
+      double UfRawValue = UfRaw();
+      if (UMaxPuReached_ == true) {
+        UfRawValue = UMaxPu_;
+      }
+      if (UMinPuReached_ == true) {
+        UfRawValue = UMinPu_;
+      }
+      if (type == ALGEBRAIC_EQ) {
```

Comment:
Please avoid duplicating the equations:
if (type != ALGEBRAIC) {
...differential equations
}
if (type != DIFFERENTIAL) {
.... algebraic equations
}